homeassistant/components/doorbird/device.py

```diff
@@ -79,43 +102,99 @@ def token(self) -> str:
-        hass = self._hass
-        # Override url if another is specified in the configuration
-        if custom_url := self.custom_url:
-            hass_url = custom_url
-        else:
-            # Get the URL of this server
-            hass_url = get_url(hass, prefer_external=False)
-
         if not self.door_station_events:
             # User may not have permission to get the favorites
             return
 
-        favorites = await self.device.favorites()
-        for event in self.door_station_events:
-            if await self._async_register_event(hass_url, event, favs=favorites):
-                _LOGGER.info(
-                    "Successfully registered URL for %s on %s", event, self.name
-                )
+        http_fav = await self._async_register_events()
+        event_config = await self._async_get_event_config(http_fav)
+        _LOGGER.debug("%s: Event config: %s", self.name, event_config)
+        if event_config.unconfigured_favorites:
+            await self._configure_unconfigured_favorites(event_config)
+            event_config = await self._async_get_event_config(http_fav)
+        self.event_descriptions = event_config.events
 
-        schedule: list[DoorBirdScheduleEntry] = await self.device.schedule()
-        http_fav: dict[str, dict[str, Any]] = favorites.get("http") or {}
-        favorite_input_type: dict[str, str] = {
+    async def _configure_unconfigured_favorites(
+        self, event_config: DoorbirdEventConfig
+    ) -> None:
+        """Configure unconfigured favorites."""
+        for entry in event_config.schedule:
+            modified_schedule = False
+            for identifier in event_config.unconfigured_favorites.get(entry.input, ()):
+                schedule = DoorBirdScheduleEntrySchedule()
+                schedule.add_weekday(MIN_WEEKDAY, MAX_WEEKDAY)
+                entry.output.append(
+                    DoorBirdScheduleEntryOutput(
+                        enabled=True,
+                        event=HTTP_EVENT_TYPE,
+                        param=identifier,
+                        schedule=schedule,
+                    )
+                )
+                modified_schedule = True
+
+            if modified_schedule:
+                update_ok, code = await self.device.change_schedule(entry)
+                if not update_ok:
+                    _LOGGER.error(
+                        "Unable to update schedule entry %s to %s. Error code: %s",
+                        self.name,
+                        entry.export,
+                        code,
+                    )
+
+    async def _async_register_events(self) -> dict[str, Any]:
+        """Register events on device."""
+        # Override url if another is specified in the configuration
+        if custom_url := self.custom_url:
+            hass_url = custom_url
+        else:
+            # Get the URL of this server
+            hass_url = get_url(self._hass, prefer_external=False)
```

Comment:
This is legacy from yaml when custom urls were supported.

It should be deprecated at some point.